apps/x509.c

```diff
@@ -1106,9 +1107,11 @@ static ASN1_INTEGER *x509_load_serial(const char *CAfile,
         memcpy(buf, CAfile, len);
         memcpy(buf + len, POSTFIX, sizeof(POSTFIX));
         serialfile = buf;
+        default_file = 1;
+        create++;
```

Comment:
Why this? The file should NOT be created if create was not set before.

Reply:
The point is that in case a default file name is used and this file does not exist, a random serial number is to be created, but this number must not be stored (i.e., the file must not be created) in case `-CAcreateserial` was not given.